### PR TITLE
[v7] Use `openpgp/lightweight` build and run tests in browser

### PR DESCRIPTION
**Pull request by larabr**
Since the `lightweight` build is intended for browsers only, this PR also changes the test suite (previously run in node):
- use `mocha` instead of `ava`
- run tests in headless browser (Chrome)
- bundle tests and app with webpack
- enable linting of test files

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -14,10 +15,13 @@
     },
     "env": {
         "es6": true,
-        "node": true
+        "node": true,
```
Comment (twiss) [inline .eslintrc.json:18]: Is the Node env still needed?

File: test/key/utils.spec.ts [hunk 2]
```diff
@@ -287,7 +201,12 @@ gm9nAQDY//xzc2hy6Efz8NqDJeLg1lh2sZkKcMXP3L+CJbhWJQEAuI6UDakE
...
\ No newline at end of file
+        const sessionKey = await generateSessionKeyFromKeyPreferences(key);
+        expect(sessionKey.data.length).to.equal(32);
+        expect(sessionKey.algorithm).to.equal('aes256');
+    });
+})
+
+
+
+
```
Comment (twiss) [inline test/key/utils.spec.ts:212]: This file is hoarding all the newlines :upside_down_face: Some of the other files need some too :cry:

File: test/setup.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import { use as chaiUse } from 'chai';
+import * as chaiAsPromised from 'chai-as-promised';
+
+import { init } from '../lib';
+
+chaiUse(chaiAsPromised);
+before(init);
```
Comment (twiss) [inline test/setup.ts:7]: E.g. this one

File: karma.conf.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        // enable / disable watching file and executing tests whenever any file changes
+        autoWatch: false,
+
+        // start these browsers
+        // available browser launchers: https://www.npmjs.com/search?q=keywords:karma-launcher
+        browsers: ['ChromeHeadless'],
+
+        // Continuous Integration mode
+        // if true, Karma captures browsers, runs the tests and exits
+        singleRun: false,
```
Comment (twiss) [inline karma.conf.js:59]: Was it intentional to have `singleRun: false` here and then `--single-run` in the package.json? It seems like you could just specify it directly?

File: test/key/processMIME.spec.ts [hunk 5]
```diff
@@ -13,46 +12,51 @@ import {
...
+            {
+                publicKeys: await getKeys(key)
+            },
+            multiPartMessageWithSpecialCharacter
+        );
+        expect(verified).to.equal(VERIFICATION_STATUS.SIGNED_AND_VALID);
+        expect(body).to.equal('hello');
+    });
+    
+})
```
Comment (twiss) [inline test/key/processMIME.spec.ts:60]: This is very nitpicky but
```suggestion
});
```
(A few other files are missing the last semicolon too)

**Review by twiss — COMMENTED** [review]
Looks good overall :+1: Small nitpicks below

File: test/key/processMIME.spec.ts (hunk 5 above)
Comment (larabr) [reply test/key/processMIME.spec.ts:60]: I'll try to have the linter enforce this + the EOF new line

**Review by twiss — APPROVED** [review]

**Pull request merged**